### ISSUE TITLE
OverviewItem Config Dialog ignore saved Filter rules in TrendsView (latest master)

### ISSUE THREAD

**Issue by PonessM** [GoldenCheetah/GoldenCheetah#4001]
Following Workflow:
  * Change Filter of a single Card in a Overview, for example to "isRun <>0"
  * Close the ConfigDialog --> Filter ist applied
  * Open the ConfigDialog --> Filter ist shown correctly
  * Restart GC 
  * Open Trendsview - Overview
  * Filter is applied correctly
  * Open ConfigDialog --> SearchBoxFilter Element is Empty

I searched in the Code, but does not find the position where Config from home-perspective.xml is load. It seems, that at the moment where "OverviewItemConfig::setWidgets()" is called, the "item->datafilter" String is empty. 

When you later switch to TrendView the "item->datafilter" is filled, so that setFilter(ChartSpaceItem *item, Specification &spec) in OverviewItems.cpp works as expected. But at this Moment, the OverviewItemConfig->filterEditor->setFilter(item->datafilter) isn't called anymore.

Hope you can follow my explanation.

Greetings Marcel

**Comment by liversedge** [comment]
Yes, basically upon restart the filter is applied correctly but if you try and edit it the filter is empty.
I saw this recently and will fix, thanks for the report.
